tools/int_info/int_info.cc

```diff
@@ -0,0 +1,199 @@
+  pcpp::RawPacket raw_packet;
+  uint32_t prev_seq_no = 0;
+  size_t flow_with_multiple_report = 0;
+  while (pcap_reader->getNextPacket(raw_packet)) {
+    total_reports++;
+    pcpp::Packet parsedPacket(&raw_packet);
+
+    pcpp::UdpLayer* udp_layer = parsedPacket.getLayerOfType<pcpp::UdpLayer>();
+    if (!udp_layer) {
+      output << "No UDP header" << std::endl;
+      skipped++;
+      continue;
+    }
+    uint16_t l4_dport = ntohs(udp_layer->getUdpHeader()->portDst);
+
+    if (l4_dport != 32766) {
+      output << "UDP port not 32766: " << l4_dport << std::endl;
+      skipped++;
+      continue;
+    }
+    pcpp::PayloadLayer* payload_layer =
+        parsedPacket.getLayerOfType<pcpp::PayloadLayer>();
+
+    // Parse INT Header
+    uint8_t* payload = payload_layer->getPayload();
+    size_t payload_len = payload_layer->getPayloadLen();
+
+    std::shared_ptr<IntFixedHeader> int_fix_report =
+        ParseIntFixedHeader(&payload, &payload_len);
+
+    if (!int_fix_report) {
+      output << "No fix report" << std::endl;
+      skipped++;
+      continue;
+    }
+
+    uint32_t seq_no = ntohl(int_fix_report->seq_no);
+    if (prev_seq_no == 0) {
+      prev_seq_no = seq_no;
+    } else {
+      if (prev_seq_no != seq_no - 1) {
+        output << "Wrong sequence number: " << seq_no << ", should be "
+               << prev_seq_no + 1 << std::endl;
+      }
+      prev_seq_no = seq_no;
+    }
+
+    std::shared_ptr<IntLocalReport> int_local_report =
+        ParseIntLocalReport(&payload, &payload_len);
+
+    if (!int_local_report) {
+      output << "No local report" << std::endl;
+      skipped++;
+      continue;
+    }
+
+    // The inner packet
+    struct timeval t = {0};
+    pcpp::RawPacket inner_packet(payload, payload_len, t, false,
+                                 pcpp::LINKTYPE_ETHERNET);
+    pcpp::Packet inner_parsed_packet(&inner_packet);
+    pcpp::IPv4Layer* inner_ipv4_layer =
+        inner_parsed_packet.getLayerOfType<pcpp::IPv4Layer>();
+    pcpp::TcpLayer* inner_tcp_layer =
+        inner_parsed_packet.getLayerOfType<pcpp::TcpLayer>();
+    pcpp::UdpLayer* inner_udp_layer =
+        inner_parsed_packet.getLayerOfType<pcpp::UdpLayer>();
+
+    if (inner_ipv4_layer) {
+      pcpp::iphdr* iph = inner_ipv4_layer->getIPv4Header();
+      uint32_t inner_src = inner_ipv4_layer->getSrcIpAddress().toInt();
+      uint32_t inner_dst = inner_ipv4_layer->getDstIpAddress().toInt();
+      uint8_t inner_proto = inner_ipv4_layer->getIPv4Header()->protocol;
+      uint16_t inner_l4_sport = 0;
+      uint16_t inner_l4_dport = 0;
+      if (inner_tcp_layer) {
+        inner_l4_sport = ntohs(inner_tcp_layer->getTcpHeader()->portSrc);
+        inner_l4_dport = ntohs(inner_tcp_layer->getTcpHeader()->portDst);
+      } else if (inner_udp_layer) {
+        inner_l4_sport = ntohs(inner_udp_layer->getUdpHeader()->portSrc);
+        inner_l4_dport = ntohs(inner_udp_layer->getUdpHeader()->portDst);
+      }
```

Comment:
I think we should have an else case here that counts the packet has skipped. If not UDP or TCP this is probably a fragment or some other garbage that shouldn't affect our measurements (since we cannot correlate the flow)

Reply:
skip packet which is not UDP or TCP now